src/mgos_barometer_bme280.c

```diff
@@ -116,17 +153,19 @@ bool mgos_barometer_bme280_read(struct mgos_barometer *dev) {
   }
 
   // read data from sensor
-  uint8_t data[6];
-  if (!mgos_i2c_read_reg_n(dev->i2c, dev->i2caddr, BME280_REG_PRESSURE_MSB, 6, data)) {
+  uint8_t data[8];
```

Comment:
I suppose the answer will be 'yes', but have you asserted that it's safe to read 8 bytes from a BMP280 which doesn't have these registers for `Hadc` ?

Reply:
In the "official" Bosch driver they do read 8 bytes at once. I would test this on a BMP280 once I can build a dev board with it.